python/cudf/cudf/io/text.py

```diff
@@ -14,17 +14,33 @@ def read_text(
     filepath_or_buffer,
     delimiter=None,
     byte_range=None,
+    compression=None,
+    compression_offsets=None,
     **kwargs,
 ):
     """{docstring}"""
 
-    filepath_or_buffer, compression = ioutils.get_reader_filepath_or_buffer(
+    if delimiter is None:
+        raise ValueError("delimiter needs to be provided")
+
+    filepath_or_buffer, _ = ioutils.get_reader_filepath_or_buffer(
         path_or_data=filepath_or_buffer,
         compression=None,
         iotypes=(BytesIO, StringIO),
         **kwargs,
     )
 
+    if compression == "bgzip":
+        if isinstance(filepath_or_buffer, TextIOBase):
+            raise ValueError("bgzip compression requires a file path")
+        filepath_or_buffer = libtext.BGZIPFile(
```

Comment:
So I understand correctly, which part feels hack?y Adding a new `BGZIPFile` type so that we can `isinstance()` it in `read_text`?

Instead, can we not just pass the `compression` and `compression_offsets` through to `cudf._lib.text.read_text`?

Reply:
Yes, that seemed a bit weird to me. Passing them through to read_text seems sensible, thanks for the suggestion! That also simplifies the remaining code a bit.